### PR TITLE
fix: missing deadline in swaps stx status screen

### PR DESCRIPTION
**Pull request by infiniteflower**
## **Description**

This PR fixes an issue where the STX status screen for a swap was showing a 0:00 for the timer.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25779?quickstart=1)

## **Related issues**

Related to: https://github.com/MetaMask/metamask-extension/pull/25063/

## **Manual testing steps**

1. Make sure Smart Transactions is on (Settings > Advanced)
2. Do a Swap
3. Observe that timer is not 0:00 and is a reasonable number

### **Before**

https://github.com/MetaMask/metamask-extension/assets/139582705/26fe6167-614f-4771-b35b-10803bc23fc0

### **After**

https://github.com/MetaMask/metamask-extension/assets/139582705/d92b933d-1011-48b4-bf04-344f275d35db

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by martahj — DISMISSED** [review]

File: test/e2e/tests/metrics/state-snapshots/errors-after-init-opt-in-ui-state.json [hunk 1]
```diff
@@ -247,6 +247,7 @@
       "swapsQuoteRefreshTime": 60000,
       "swapsQuotePrefetchingRefreshTime": 60000,
       "swapsStxBatchStatusRefreshTime": 10000,
+      "swapsStxStatusDeadline": "number",
```
Comment (dan437) [inline test/e2e/tests/metrics/state-snapshots/errors-after-init-opt-in-ui-state.json:250]: Why is this a "number" string and not some exact number like other fallback values?

**Review by dan437 — COMMENTED** [review]
Thanks for adding the fallback! One thing: this PR adds a fallback for the deadline, but it should use a value from backend in the first place. Can you check if the value from backend is being used?

**Review by digiwand — DISMISSED** [review]

**Comment by martahj** [comment]
> Thanks for adding the fallback! One thing: this PR adds a fallback for the deadline, but it should use a value from backend in the first place. Can you check if the value from backend is being used?

@dan437 You were right - the fallback rather than the value from the PR was used. I pushed some updates and that is no longer the case.

File: test/e2e/tests/metrics/state-snapshots/errors-after-init-opt-in-ui-state.json (hunk 1 above)
Comment (martahj) [reply test/e2e/tests/metrics/state-snapshots/errors-after-init-opt-in-ui-state.json]: The value wasn't being set up correctly in the test, but it is now 👍

**Review by dan437 — APPROVED** [review]

**Review by martahj — APPROVED** [review]

**Pull request merged**